org.eclipse.xtext.ide/src/org/eclipse/xtext/ide/server/LanguageServerImpl.xtend

```diff
@@ -706,7 +708,11 @@ import org.eclipse.xtext.findReferences.IReferenceFinder.IResourceAccess
 		override getLanguageClient() {
 			client
 		}
-		
+
+		override doReadIndex(Consumer<IResourceDescription> consumer) {
+			workspaceManager.index.allResourceDescriptions.forEach(consumer)
```

Comment:
This should run within `requestManager.runRead [ cancelIndicator | ... ]`.
Also it would be good to provide access to the full `IResourceDescriptions` and return a `CompletableFuture<T>`. So it is symmetric.